src/NuGet.Core/NuGet.Commands/SourcesCommands/SourceRunners.cs

```diff
@@ -286,10 +245,10 @@ public static void Run(UpdateSourceArgs args, Func<ILogger> getLogger)
                 existingSource = new Configuration.PackageSource(args.Source, existingSource.Name);
                 existingSource.AllowInsecureConnections = args.AllowInsecureConnections;
 
-                // If the existing source is not http, warn the user
-                if (existingSource.IsHttp && !existingSource.IsHttps && !existingSource.AllowInsecureConnections)
+                // If the new source is not http, throw an error
+                if (existingSource.IsHttp && !existingSource.IsHttps && !existingSource.AllowInsecureConnections && !existingSource.AllowInsecureConnections)
                 {
-                    getLogger().LogWarning(string.Format(CultureInfo.CurrentCulture, Strings.Warning_HttpServerUsage, "update source", args.Source));
+                    throw new ArgumentException(string.Format(CultureInfo.CurrentCulture, Strings.Error_HttpSource_Single, "source update", args.Source));
```

Comment:
the command name is `dotnet nuget update source` should we continue using the old code i.e. `update source`?